### ISSUE TITLE
onchange is being hit on first load?

### ISSUE THREAD

**Issue by joshuataylor** [RyanNerd/ember-power-timepicker#11]
When running the example, it looks like onchange is being hit on first load?

Is this normal?

**Comment by RyanNerd** [comment]
Yes. Unfortunately this is deliberate. Because ember-power-time picker is _subclassing_ ember-power-select by inserting ember-power-select in the DOM some setup can't be done in the `init()` because the component itself hasn't yet been added to the DOM. The code causing this is [here](https://github.com/RyanNerd/ember-power-timepicker/blob/b32bb45aa7d18f2cf504d416784b203ed67506b8/addon/components/time-picker.js#L440):

```
    didInsertElement()
    {
      // If selectedTime is set/overridden/calculated we need to tell power-select about it.
      if (this.get('selectedTime') !== null) {
        // We can't do this in the init() because power-select has not been loaded into the DOM yet.
        try {
          this.propertyWillChange('selectedTime');
          this.propertyDidChange('selectedTime');
        } catch (e) {
          // Ugly work-around by setting selectedTime to itself.
          this.set('selectedTime', this.get('selectedTime'));
        }
      }
    }
```
If you are aware of a better method or a better work-around please submit a PR.

**Comment by joshuataylor** [comment]
Ah cool, not sure of a workaround yet either.

One problem with this is that your model gets changed, and if you have something happen on change, it's an endless loop.

I'll also try to think of a solution.
